reflex/state.py

```diff
@@ -611,17 +648,17 @@ def __setattr__(self, name: str, value: Any):
         # Add the var to the dirty list.
         if name in self.vars or name in self.computed_var_dependencies:
             self.dirty_vars.add(name)
-            self.mark_dirty()
+            self._mark_dirty()
 
         # For now, handle router_data updates as a special case
         if name == constants.ROUTER_DATA:
             self.dirty_vars.add(name)
-            self.mark_dirty()
+            self._mark_dirty()
             # propagate router_data updates down the state tree
             for substate in self.substates.values():
                 setattr(substate, name, value)
 
-    def reset(self):
+    def _reset(self):
```

Comment:
I know some people are using the `.reset()` function in their apps already. Should we maybe keep this one as is for now? We throw a warning anyways, so it can make sure people don't accidentally override it